pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -235,10 +235,68 @@ var logger *logrus.Logger
 var log *base.LogObject
 var zedcloudCtx *zedcloud.ZedCloudContext
 
+type infoDest uint
+
+// destination types, where info should be sent
+const (
+	ControllerDest  infoDest = 1
+	LocalServerDest          = 2
+	LOCDest                  = 4
+	AllDest                  = ControllerDest | LocalServerDest | LOCDest
+)
+
+func ignoreErrFlags(dest infoDest, ignoreErr infoDest) zedcloud.DeferredItemFlags {
+	flags := zedcloud.DeferredItem_NoFlags
+	if ignoreErr&dest != 0 {
+		// Ignore errors so that the item be handled with highest priority
+		// out-of-band and not be affected by stalled requests in the normal
+		// queue. See the getDeferredPriorityFunctions() for details how the
+		// @DeferredItem_IgnoreError is handled.
+		flags |= zedcloud.DeferredItem_IgnoreError
+	}
+	return flags
+}
+
+func createDeferredItemsByDest(ctx *zedagentContext, dest infoDest,
+	ignoreErr infoDest, key string, buf *bytes.Buffer, size int64,
+	bailOnHTTPErr, withNetTracing bool, itemType interface{}) []*zedcloud.DeferredItem {
+
+	locConfig := ctx.getconfigCtx.locConfig
+	items := []*zedcloud.DeferredItem{}
+
+	// Be aware: key is modified in order to have a url as a postfix.
+	//           This is needed in order to send similar data to
+	//           different destinations (urls).
+	//
+	//           If you need to remove the deferred item afterwards by
+	//           the exact key - probably this is not the function which
+	//           you need.
+
+	if dest&ControllerDest != 0 {
+		url := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API,
+			devUUID, "info")
+		flags := ignoreErrFlags(ControllerDest, ignoreErr)
+		item := zedcloud.CreateDeferred(key+url, buf, size, url, bailOnHTTPErr,
+			withNetTracing, itemType, flags)
+		items = append(items, item)
+	}
+	if dest&LOCDest != 0 && locConfig != nil {
+		url := zedcloud.URLPathString(locConfig.LocURL, zedcloudCtx.V2API,
+			devUUID, "info")
+		flags := ignoreErrFlags(LOCDest, ignoreErr)
+		item := zedcloud.CreateDeferred(key+url, buf, size, url, bailOnHTTPErr,
+			withNetTracing, itemType, flags)
+		items = append(items, item)
+	}
+
```

Comment:
Do you plan to later also implement `LocalServerDest` here?

Reply:
Not now. Did not want to touch this part.

Reply:
But you already have AllDest include LocalServerDest yet you only process that for the DeviceInfo. So it makes it hard to understand what does AllDest mean since it is type dependent what the code actually does.